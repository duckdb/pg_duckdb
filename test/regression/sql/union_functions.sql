-- Setup
select duckdb.raw_query('CREATE TABLE union_tbl1 (u UNION(num INTEGER, str VARCHAR));');
select duckdb.raw_query('INSERT INTO union_tbl1 VALUES (1), (''two''), (union_value(str := ''three''));');

select * from duckdb.query($$ Select * from union_tbl1 $$);

select union_tag(r['u']) from duckdb.query($$ Select u from union_tbl1 $$) r;

select union_extract(r['u'],'str') from duckdb.query($$ Select u from union_tbl1 $$) r;
select union_extract(r['u'],'num') from duckdb.query($$ Select * from union_tbl1 $$) r;

SELECT union_extract(r['u'], 'str') FROM duckdb.query($$ Select u from union_tbl1 $$) r WHERE union_tag(r['u']) = 'str';
SELECT union_extract(r['u'], 'str') FROM duckdb.query($$ Select u from union_tbl1 $$) r WHERE union_extract(r['u'], 'str') IS NOT NULL;

select duckdb.raw_query('DROP TABLE union_tbl1;');
