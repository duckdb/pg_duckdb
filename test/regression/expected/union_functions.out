-- Setup
select duckdb.raw_query('CREATE TABLE union_tbl1 (u UNION(num INTEGER, str VARCHAR));');
NOTICE:  result: Count	
BIGINT	
[ Rows: 0]


 raw_query 
-----------
 
(1 row)

select duckdb.raw_query('INSERT INTO union_tbl1 VALUES (1), (''two''), (union_value(str := ''three''));');
NOTICE:  result: Count	
BIGINT	
[ Rows: 1]
3


 raw_query 
-----------
 
(1 row)

select * from duckdb.query($$ Select * from union_tbl1 $$);
   u   
-------
 1
 two
 three
(3 rows)

select union_tag(r['u']) from duckdb.query($$ Select u from union_tbl1 $$) r;
 union_tag 
-----------
 num
 str
 str
(3 rows)

select union_extract(r['u'],'str') from duckdb.query($$ Select u from union_tbl1 $$) r;
 union_extract 
---------------
 
 two
 three
(3 rows)

select union_extract(r['u'],'num') from duckdb.query($$ Select * from union_tbl1 $$) r;
 union_extract 
---------------
             1
              
              
(3 rows)

SELECT union_extract(r['u'], 'str') FROM duckdb.query($$ Select u from union_tbl1 $$) r WHERE union_tag(r['u']) = 'str';
 union_extract 
---------------
 two
 three
(2 rows)

SELECT union_extract(r['u'], 'str') FROM duckdb.query($$ Select u from union_tbl1 $$) r WHERE union_extract(r['u'], 'str') IS NOT NULL;
 union_extract 
---------------
 two
 three
(2 rows)

select duckdb.raw_query('DROP TABLE union_tbl1;');
NOTICE:  result: Success	
BOOLEAN	
[ Rows: 0]


 raw_query 
-----------
 
(1 row)

